src/main/java/alfio/manager/ExtensionManager.java

```diff
@@ -353,35 +351,49 @@ public boolean handlePdfTransformation(String html, Event event, OutputStream ou
     }
 
 
-    private void asyncCall(ExtensionEvent extensionEvent, Event event, Map<String, Object> payload) {
+    private void asyncCall(ExtensionEvent extensionEvent, Purchasable event, Map<String, Object> payload) {
         extensionService.executeScriptAsync(extensionEvent.name(),
-            toPath(event.getOrganizationId(), event.getId()),
+            toPath(event),
             fillWithBasicInfo(payload, event));
     }
 
-    private <T> T syncCall(ExtensionEvent extensionEvent, Event event, Map<String, Object> payload, Class<T> clazz) {
+    private <T> T syncCall(ExtensionEvent extensionEvent, Purchasable purchasable, Map<String, Object> payload, Class<T> clazz) {
         try {
             return extensionService.executeScriptsForEvent(extensionEvent.name(),
-                toPath(event.getOrganizationId(), event.getId()),
-                fillWithBasicInfo(payload, event),
+                toPath(purchasable),
+                fillWithBasicInfo(payload, purchasable),
                 clazz);
         } catch(AlfioScriptingException ex) {
             log.warn("Unexpected exception while executing script:", ex);
             return null;
         }
     }
 
-    private Map<String, Object> fillWithBasicInfo(Map<String, Object> payload, Event event) {
+    private Map<String, Object> fillWithBasicInfo(Map<String, Object> payload, Purchasable purchasable) {
         Map<String, Object> payloadCopy = new HashMap<>(payload);
-        payloadCopy.put("event", event);
-        payloadCopy.put("eventId", event.getId());
-        payloadCopy.put("organizationId", event.getOrganizationId());
+        //FIXME ugly
+        if (purchasable instanceof Event) {
+            var event = (Event) purchasable;
+            payloadCopy.put("event", event);
+            payloadCopy.put("eventId", event.getId());
+        } else {
```

Comment:
```suggestion
        
```
always include the "purchasable" object, so that the scripts can be migrated

Reply:
fixed